src/ui/public/filter_bar/lib/add_filters_to_kuery.js

```diff
@@ -0,0 +1,39 @@
+import _ from 'lodash';
+import { FilterBarLibMapAndFlattenFiltersProvider } from 'ui/filter_bar/lib/map_and_flatten_filters';
+import { FilterBarLibExtractTimeFilterProvider } from 'ui/filter_bar/lib/extract_time_filter';
+import { FilterBarLibChangeTimeFilterProvider } from 'ui/filter_bar/lib/change_time_filter';
+import { FilterBarLibFilterOutTimeBasedFilterProvider } from 'ui/filter_bar/lib/filter_out_time_based_filter';
+import { toKueryExpression, fromKueryExpression, nodeTypes, filterToKueryAST } from 'ui/kuery';
+
+export function AddFiltersToKueryProvider(Private) {
+  const mapAndFlattenFilters = Private(FilterBarLibMapAndFlattenFiltersProvider);
+  const extractTimeFilter = Private(FilterBarLibExtractTimeFilterProvider);
+  const changeTimeFilter = Private(FilterBarLibChangeTimeFilterProvider);
+  const filterOutTimeBasedFilter = Private(FilterBarLibFilterOutTimeBasedFilterProvider);
+
+  return async function addFiltersToKuery(state, filters) {
```

Comment:
This function is based on `addLegacyFilter` from the query manager. Moved here for use in the filter_bar_click_handler.